libs/sdk-py/langgraph_sdk/encryption/__init__.py

```diff
@@ -0,0 +1,454 @@
+        """Register a blob encryption handler.
+
+        The handler will be called to encrypt opaque data like checkpoint blobs.
+
+        Example:
+            ```python
+            @encrypt.blob
+            async def encrypt_blob(ctx: EncryptionContext, blob: bytes) -> bytes:
+                # Encrypt the blob using your encryption service
+                return encrypted_blob
+            ```
+
+        Args:
+            fn: The encryption handler function
+
+        Returns:
+            The registered handler function
+        """
+        self._parent._blob_encryptor = fn
+        return fn
+
+    @property
+    def json(self) -> _JsonEncryptDecorators:
+        """Access JSON encryption decorators.
+
+        Supports model-specific handlers:
+        - @encrypt.json - default handler for all models
+        - @encrypt.json.thread - handler for thread model only
+        - @encrypt.json.assistant - handler for assistant model only
+        - @encrypt.json("custom") - handler for custom model name
+
+        Example:
+            ```python
+            @encrypt.json
+            async def default_encrypt(ctx: EncryptionContext, data: dict) -> dict:
+                # Default encryption for all models
+                return encrypt_data(data)
+
+            @encrypt.json.thread
+            async def encrypt_thread(ctx: EncryptionContext, data: dict) -> dict:
+                # Special encryption for thread model only
+                return encrypt_thread_data(data)
+            ```
+        """
+        return self._json
+
+
+class _DecryptDecorators:
+    """Decorators for decryption handlers.
+
+    Provides @decrypt.blob and @decrypt.json decorators for
+    registering decryption functions.
+    """
+
+    def __init__(self, parent: Encrypt):
+        self._parent = parent
+        self._json = _JsonDecryptDecorators(parent)
+
+    def blob(self, fn: types.BlobDecryptor) -> types.BlobDecryptor:
+        """Register a blob decryption handler.
+
+        The handler will be called to decrypt opaque data like checkpoint blobs.
+
+        Example:
+            ```python
+            @decrypt.blob
+            async def decrypt_blob(ctx: EncryptionContext, blob: bytes) -> bytes:
+                # Decrypt the blob using your encryption service
+                return decrypted_blob
+            ```
+
+        Args:
+            fn: The decryption handler function
+
+        Returns:
+            The registered handler function
+        """
+        self._parent._blob_decryptor = fn
+        return fn
+
+    @property
+    def json(self) -> _JsonDecryptDecorators:
+        """Access JSON decryption decorators.
+
+        Supports model-specific handlers:
+        - @decrypt.json - default handler for all models
+        - @decrypt.json.thread - handler for thread model only
+        - @decrypt.json.assistant - handler for assistant model only
+        - @decrypt.json("custom") - handler for custom model name
```

Comment:
ooc: what's the use case for "custom" ?

Reply:
idk, i just wrote it first before i figured out that the other technique could be dynamically dispatched. will remove.